ui/src/views/image/RegisterOrUploadIso.vue

```diff
@@ -467,6 +478,12 @@ export default {
       this.fileList = newFileList
       this.form.file = undefined
     },
+    handleDirectDownloadChange () {
+      if (this.form.zoneid && this.zoneList.find(entry => entry.id === this.form.zoneid)) {
+        return
+      }
+      this.form.zoneid = this.zoneList?.[0]?.id || ''
```

Comment:
@shwstppr 
line 485 set the this.form.zoneid to the id of first zone , is it needed ?

Reply:
@weizhouapache yes. This is for the case when the user selects direct-download, then selects an edge zone but then changes direct-download to false. Earlier selected zone will not be available now se the code selects the first zone from the list